src/version.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Utility class, useful for representing and manipulating version strings (e.g. of IDE dependencies).
+ */
+export class Version {
+    readonly major: number;
+    readonly minor: number;
+    readonly patch: number;
+
+    constructor(major: number, minor: number, patch: number) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+
+    static unspecified(): Version {
+        return new Version(0, 0, 0);
+    }
+
+    static parse(versionString: string) {
+        // Only keep numbers and dots.
+        let normalizedVersionString = versionString.replace(/[^\d.]/g, "");
+        let [major, minor, patch] = normalizedVersionString.split(".");
+        let majorNumber = parseInt(major);
+        let minorNumber = parseInt(minor);
+        let patchNumber = parseInt(patch);
+
+        if (isNaN(majorNumber) || isNaN(minorNumber) || isNaN(patchNumber)) {
+            throw new CannotParseVersionError(versionString);
+        }
+
+        return new Version(majorNumber, minorNumber, patchNumber);
+    }
+
+    isNewerOrSameAs(other: Version) {
+        return this.major >= other.major && this.minor >= other.minor && this.patch >= other.patch;
+    }
```

Comment:
The condition is wrong - for example `2.1.0` is newer than `1.2.3`, but for this example the `isNewerOrSameAs` function would return `false`.

Reply:
I think we could use a 3rd party library for the versioning - something like https://www.npmjs.com/package/semver

Reply:
Oh, :facepalm: Thanks :pray: 

Will fix.

Reply:
Indeed, using `semver` is a good idea. However, is it OK to not use it yet? 
Fixed the comparison.